### ISSUE TITLE
Wrong type of defaultValue

### ISSUE THREAD

**Issue by enumag** [overblog/GraphQLBundle#399]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Version/Branch   | 0.11.7

I've changed my setup to declare most of my schema using [GraphQL schema language](https://github.com/overblog/GraphQLBundle/blob/master/docs/definitions/graphql-schema-language.md) and I implement a few customized ResolverMaps to pull the data from the underlying database.

This works ok but encountered a problem with type like this:

```gql
input Paging {
  page: Int! = 1
  pageSize: Int!
}
```

Here is the generated type:

```php
<?php
namespace App\Generated\GraphQL;

use GraphQL\Type\Definition\InputObjectType;
use GraphQL\Type\Definition\Type;
use Overblog\GraphQLBundle\Definition\ConfigProcessor;
use Overblog\GraphQLBundle\Definition\LazyConfig;
use Overblog\GraphQLBundle\Definition\GlobalVariables;
use Overblog\GraphQLBundle\Definition\Type\GeneratedTypeInterface;

/**
 * THIS FILE WAS GENERATED AND SHOULD NOT BE MODIFIED!
 */
final class PagingType extends InputObjectType implements GeneratedTypeInterface
{

    public function __construct(ConfigProcessor $configProcessor, GlobalVariables $globalVariables = null)
    {
        $configLoader = function(GlobalVariables $globalVariable) {
            return [
            'name' => 'Paging',
            'description' => null,
            'fields' => function () use ($globalVariable) {
                return [
                'page' => [
                    'type' => Type::nonNull(Type::int()),
                    'description' => null,
                    'defaultValue' => '1',
                ],
                'pageSize' => [
                    'type' => Type::nonNull(Type::int()),
                    'description' => null,
                ],
            ];
            },
        ];
        };
        $config = $configProcessor->process(LazyConfig::create($configLoader, $globalVariables))->load();
        parent::__construct($config);
    }
}
```

The problem is that the defaultValue is `'1'` (string) instead of `1` (int) because I then receive a string value where I normally expect an int value.

After some digging I think this should be fixed in `GraphQLParser::astValueNodeToConfig` (or `FieldsNode::astValueNodeToConfig` in master) with something like this:

```php
        switch ($valueNode->kind) {
            case NodeKind::INT:
            	$config = (int) $valueNode->value;
            	break;
            	
            case NodeKind::FLOAT:
	            $config = (float) $valueNode->value;
	            break;

	        case NodeKind::BOOLEAN:
	        case NodeKind::STRING:
	        case NodeKind::ENUM:
                $config = $valueNode->value;
                break;
            // ...
```

Should I send a PR? Or is there some reason for the current behavior?

Another possible place to fix it could be the `overblog/graphql-php-generator` package but I think it's better to have the correct type in the extension configuration too.

**Comment by enumag** [comment]
Or maybe we should use `AST::valueFromASTUntyped()` instead?

https://github.com/webonyx/graphql-php/blob/master/src/Utils/AST.php#L507

**Comment by mcg-web** [comment]
Hi @enumag thanks for bringing this up, this effectively a bug. Using native AST method is maybe the best way to go. This should be fix from 0.11 branch. Do you want to submit a PR to fix it please?

**Comment by mcg-web** [comment]
Thank you again
